src/Shared/FrameworkLocationHelper.cs

```diff
@@ -145,7 +145,7 @@ internal static class FrameworkLocationHelper
         /// <summary>
         /// List the supported .net versions.
         /// </summary>
-        private static readonly DotNetFrameworkSpec[] s_dotNetFrameworkSpecs =
+        private static readonly DotNetFrameworkSpec[] DotNetFrameworkSpecs =
```

Comment:
Can you please make this one and `VisualStudioSpecs` lazy as well? Likely by just refactoring them into private static methods instead of fields. The methods would be called from the `Lazy` lambdas you have added.

Reply:
This code relies on the VisualStudioSpecs being an array so it can find the previous version by index. So VisualStudioSpecs can't be refactored into a static method. But it can be lazily loaded.
https://github.com/dotnet/msbuild/blob/daa103ebba42cb56ba18caefa74399a848905d3d/src/Shared/FrameworkLocationHelper.cs#L1472-L1484